desc/vmec_utils.py

```diff
@@ -468,3 +482,451 @@ def vmec_boundary_subspace(eq, RBC=None, ZBS=None, RBS=None, ZBC=None):  # noqa:
+    grid = LinearGrid(M=eq.M_grid, N=eq.N_grid, rho=r_half, NFP=NFP, sym=eq.sym)
+    data_1d = eq.compute(keys_1d, grid=grid)
+    jlist = file.createVariable("jlist", np.int32, ("comput_surfs",))
+    jlist.long_name = (
+        "1-based radial indices of the surfaces for which the "
+        "transformation to Boozer coordinates was computed. 2 corresponds to the "
+        "first half-grid point."
+    )
+    jlist.units = "None"
+    jlist[:] = np.arange(1, s_half.size + 1) + 1
+
+    ixm_b = file.createVariable("ixm_b", np.int32, ("mn_modes",))
+    ixm_b.long_name = (
+        "Poloidal mode numbers m for which the Fourier amplitudes rmnc,"
+        "bmnc etc are stored"
+    )
+    ixm_b.units = "None"
+    ixm_b[:] = modes[:, 1] if eq.sym else modes[inds_cos, 1]
+
+    ixn_b = file.createVariable("ixn_b", np.int32, ("mn_modes",))
+    ixn_b.long_name = (
+        "Toroidal mode numbers n for which the Fourier amplitudes"
+        "rmnc, bmnc etc are stored"
+    )
+    ixn_b.units = "None"
+    ixn_b[:] = modes_sin[:, 2] * eq.NFP if eq.sym else modes[inds_cos, 2] * eq.NFP
+
+    iotas = file.createVariable("iota_b", np.float64, ("radius",))
+    iotas.long_name = (
+        "Rotational transform. The radial grid corresponds to the"
+        "requested surfaces, and a 0 is prepended"
+    )
+    iotas.units = "None"
+
+    iotas[0:] = np.insert(
+        -grid.compress(eq.compute("iota", grid=grid, data=data_1d)["iota"]), 0, 0
+    )
+
+    buco_b = file.createVariable("buco_b", np.float64, ("radius",))
+    buco_b.long_name = (
+        "Coefficient multiplying grad theta_Boozer in the covariant"
+        "representation of the magnetic field vector, often denoted I(psi)."
+    )
+    buco_b.units = "None"
+    buco_b[0] = 0
+
+    buco_b[1:] = -grid.compress(data_1d["I"])
+
+    bvco_b = file.createVariable("bvco_b", np.float64, ("radius",))
+    bvco_b.long_name = (
+        "Coefficient multiplying grad zeta_Boozer in the covariant"
+        "representation of the magnetic field vector, often denoted G(psi)."
+    )
+    bvco_b.units = "None"
+    bvco_b[0] = 0
+
+    bvco_b[1:] = grid.compress(data_1d["G"])
+
+    # TODO: assuming this is on full mesh
+    presf = file.createVariable("pres_b", np.float64, ("radius",))
+    presf.long_name = "pressure on full mesh"
+    presf.units = "Pa"
+    presf[:] = eq.compute("p", grid=LinearGrid(rho=r_full, theta=0, zeta=0))["p"]
+
+    beta = file.createVariable("beta_b", np.float64, ("radius",))
+    beta.long_name = "Blank, only included for compatibility reasons"
+    beta.units = "None"
+    beta[:] = np.zeros_like(r_full)
+
+    phipf = file.createVariable("phip_b", np.float64, ("radius",))
+    phipf.long_name = "d(phi)/ds: toroidal flux derivative, not normalized by 2pi"
+    phipf[:] = Psi * np.ones((surfs,))
+
+    phi = file.createVariable("phi_b", np.float64, ("radius",))
+    phi.long_name = "toroidal flux"
+    phi.units = "Wb"
+    phi[:] = np.linspace(0, Psi, surfs)
```

Comment:
is this supposed to be on the full or half grid?

Reply:
neither techincaly since it can be different from the chosen outputted surfaces in the boozmn, I just chose it as the full grid.
https://github.com/hiddenSymmetries/booz_xform/blob/2ceb1d7b688bf9e1a88847dfbf34c990ddd086e1/src/_booz_xform/write_boozmn.cpp#L100